homeassistant/components/diagnostics/__init__.py

```diff
@@ -195,20 +222,21 @@ async def get(
         except ValueError:
             return web.Response(status=HTTPStatus.BAD_REQUEST)
 
-        hass = request.app["hass"]
+        hass: HomeAssistant = request.app["hass"]
 
         if (config_entry := hass.config_entries.async_get_entry(d_id)) is None:
             return web.Response(status=HTTPStatus.NOT_FOUND)
 
-        if (info := hass.data[DOMAIN].get(config_entry.domain)) is None:
+        diagnostics_data: DiagnosticsData = hass.data[DOMAIN]
+        if (info := diagnostics_data.platforms.get(config_entry.domain)) is None:
             return web.Response(status=HTTPStatus.NOT_FOUND)
 
         filename = f"{config_entry.domain}-{config_entry.entry_id}"
 
         if sub_type is None:
-            if info[d_type.value] is None:
+            if info.config_entry_diagnostics is None:
```

Comment:
Why do we know that this is the config entry diagnostics type instead of using the `dtype` variable?

Reply:
We do this above
```py
        try:
            d_type = DiagnosticsType(d_type)
        except ValueError:
            return web.Response(status=HTTPStatus.BAD_REQUEST)
```
Since `DiagnosticsType` enum only has a single member, we know the request is to get config entry diagnostics.

Reply:
We *could* of course be a bit more explicit and map the value of `d_type` to the right data, but it's completely useless code as long as there's only a single possible `DiagnosticsType`, so I think it's better to add such code when it's actually needed 🤷

Reply:
There's only this view. Both config entry and device entry kinds will use this view. I think the reason is that sub_type is None.

Reply:
This code can be made clearer I think. But not in this PR.

Reply:
Yeah, sure, it's the combination of both:
- `sub_type` (which can only be `None` or `DiagnosticsSubType.DEVICE`) is `None`
- `d_type` can only be `DiagnosticsType.CONFIG_ENTRY`
- -> The request is to det config entry diagnostics